util/src/overlaydb.rs

```diff
@@ -14,19 +14,26 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
-//! Disk-backed `HashDB` implementation.
-
-use error::*;
-use hash::*;
-use bytes::*;
-use rlp::*;
-use hashdb::*;
-use memorydb::*;
-use std::ops::*;
-use std::sync::*;
-use std::env;
-use std::collections::HashMap;
+//! Disk-backed, `HashDB` implementation.
+
+use error::{BaseDataError, UtilError};
 use kvdb::{Database, DBTransaction};
+use memorydb::MemoryDB;
+use hash::{FixedHash, H32, H256};
+use hashdb::HashDB;
+use Bytes;
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+/// How to treat entries which can be deleted
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum DeletionMode {
+	/// Ignore them.
+	Ignore,
+	/// Delete them.
+	Delete,
+}
```

Comment:
ArchiveDB ignores "deletions", but other usecases for this abstraction may prefer to actually delete data from the backing database.